src/vs/workbench/api/node/extHostTextEditor.ts

```diff
@@ -435,6 +435,10 @@ export class ExtHostTextEditor implements vscode.TextEditor {
 		this._trySetSelection();
 	}
 
+	setLanguageById(language_id: string): void {
+		this._runOnProxy(() => this._proxy.$trySetLanguageById(this._id, language_id));
+	}
```

Comment:
We need to think where we expose this as API. The catch is that the API treats a mode change as a document-close/document-open-cycle. That means having that function on `TextDocument` isn't a good idea (calling a method shouldn't make the receiver disappear). The same is true for editors - for the API an editor a document (instance) that's currently visible in an editor. So, may expose this as a new function in the `language` namespace.

Reply:
To expose the API, make the change in `vscode.d.ts` or (better) `vscode.proposed.d.ts` and then fix the compile error in `extHost.api.impl.ts`

Reply:
I totally missed this part because I've tested this with a javascript - based plugin which, unlike typescript - based, doesn't need the interface.